content/en/real_user_monitoring/guide/monitor-your-nextjs-app-with-rum.md

```diff
@@ -0,0 +1,211 @@
+   [2]: https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required
+   [3]: https://nextjs.org/docs/pages
+   [4]: https://nextjs.org/docs/pages/building-your-application/routing/custom-app#usage
+   [5]: https://nextjs.org/docs/messages/next-script-for-ga#using-analyticsjs-legacy
+
+   {{% /tab %}}
+   {{% tab "CDN sync" %}}
+
+   If using CDN async, you need to make a few small changes to the code snippet from the Datadog RUM UI before pasting it into either the root `layout.tsx` or custom `_app.tsx` file (Datadog supports both):
+
+   - If your application relies on the **newer** Next.js [App Router][1] (versions 13+), paste the snippet into the root [`layout.tsx`][2] file
+   - If your Next.js application relies on the **older** Next.js [Page Router][3], paste the snippet into the custom [`_app.tsx`][4] file.
+   - The Next.js external scripts need to be loaded like in [this page][5].
+
+   **Note**: The Next.js scripts include import and export lines, and includes curly braces and backticks between the `Script id`, where all instances of `Script` are in uppercase.
+
+   {{< code-block lang="javascript" filename="layout.tsx or _app.tsx" disable_copy="false" collapsible="true" >}}
+
+   // Paste the CDN sync snippet here without first <script> and last </script> lines like below
+
+   import Script from 'next/script'
+    
+   function Home() {
+     return (
+       <div className="container">
+         <Script id="datadog-rum">
+
+         <Script
+             src="https://www.datadoghq-browser-agent.com/us1/v4/datadog-rum.js"
+             type="text/javascript">
+         </Script>
+
+             window.DD_RUM && window.DD_RUM.init({
+               clientToken: '<CLIENT_TOKEN>',
+               applicationId: '<YOUR_APPLICATION_ID',
+               site: 'datadoghq.com',
+               service: 'nextjs-test',
+               env: '<ENV_NAME>',
+               // Specify a version number to identify the deployed version of your application in Datadog 
+               // version: '1.0.0', 
+               sessionSampleRate: 100,
+               sessionReplaySampleRate: 20,
+               trackUserInteractions: true,
+               trackResources: true,
+               trackLongTasks: true,
+               defaultPrivacyLevel: 'mask-user-input',
+             });
+
+             window.DD_RUM &&
+             window.DD_RUM.startSessionReplayRecording();
+
+         </Script>
+             </div>
+           )
+         }
+          
+         export default Home
+
+      {{< /code-block >}}
+
+   [1]: https://nextjs.org/docs/app
+   [2]: https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required
+   [3]: https://nextjs.org/docs/pages
+   [4]: https://nextjs.org/docs/pages/building-your-application/routing/custom-app#usage
+   [5]: https://nextjs.org/docs/messages/next-script-for-ga#using-analyticsjs-legacy
+
+   {{% /tab %}}
+   {{< /tabs >}}
```

Comment:
This is a working example for setting up CDN sync with the app router in a root layout.tsx:
```tsx
import Script from "next/script";

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <body>
        <Script
          id="dd-rum-sync"
          src="https://www.datadoghq-browser-agent.com/us1/v4/datadog-rum.js"
          type="text/javascript"
          strategy="beforeInteractive"
        />
        <Script id="datadog-rum">
          {`
            window.DD_RUM && window.DD_RUM.init({
              clientToken: '<CLIENT_TOKEN>',
              applicationId: '<YOUR_APPLICATION_ID>',
              site: 'datadoghq.com',
              service: 'rum-cdn-async',
              env: '<ENV_NAME>',
              // Specify a version number to identify the deployed version of your application in Datadog 
              // version: '1.0.0', 
              sessionSampleRate: 100,
              sessionReplaySampleRate: 20,
              trackUserInteractions: true,
              trackResources: true,
              trackLongTasks: true,
              defaultPrivacyLevel: 'mask-user-input',
            });

            window.DD_RUM &&
            window.DD_RUM.startSessionReplayRecording();
          `}
        </Script>
        {children}
      </body>
    </html>
  );
}
```